File: go-sdk/pkg/models/shared/balancescursorresponse.go
package shared

type BalancesCursorResponseCursor struct {
	Data     []map[string]map[string]int64 `json:"data"`
	HasMore  bool                          `json:"hasMore"`
	Next     *string                       `json:"next,omitempty"`
	PageSize int64                         `json:"pageSize"`
	Previous *string                       `json:"previous,omitempty"`
}

type BalancesCursorResponse struct {
	Cursor BalancesCursorResponseCursor `json:"cursor"`
}

File: ts-sdk/src/sdk/models/shared/balancescursorresponse.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class BalancesCursorResponseCursor extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data: Record<string, Record<string, number>>[];

  @SpeakeasyMetadata({ data: "json, name=hasMore" })
  hasMore: boolean;

  @SpeakeasyMetadata({ data: "json, name=next" })
  next?: string;

  @SpeakeasyMetadata({ data: "json, name=pageSize" })
  pageSize: number;

  @SpeakeasyMetadata({ data: "json, name=previous" })
  previous?: string;
}

export class BalancesCursorResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cursor" })
  cursor: BalancesCursorResponseCursor;
}